public/components/common/modals/__tests__/model_version_delete_confirm_modal.test.tsx

```diff
@@ -0,0 +1,165 @@
+    setup();
+
+    await userEvent.type(screen.getByLabelText('confirm text input'), 'model1 version 1');
+
+    expect(modelDeleteMock).not.toHaveBeenCalled();
+    await userEvent.click(screen.getByText('Delete version'));
+    expect(modelDeleteMock).toHaveBeenCalledWith('1');
+
+    modelDeleteMock.mockRestore();
+  });
+
+  it('should show delete success toast and call closeModal with true after delete success and can not be searched', async () => {
+    const modelDeleteMock = jest.spyOn(ModelVersion.prototype, 'delete').mockResolvedValue({});
+    const modelSearchMock = jest
+      .spyOn(ModelVersion.prototype, 'search')
+      .mockResolvedValue({ data: [], total_model_versions: 0 });
+    const openSearchDashboardsMock = mockUseOpenSearchDashboards();
+    const { closeModalMock } = setup();
+
+    await userEvent.type(screen.getByLabelText('confirm text input'), 'model1 version 1');
+
+    expect(closeModalMock).not.toHaveBeenCalled();
+    await userEvent.click(screen.getByText('Delete version'));
+    await waitFor(() => {
+      expect(screen.getByTestId('euiToastHeader')).toHaveTextContent(
+        'model1 version 1 has been deleted'
+      );
+    });
+    expect(closeModalMock).toHaveBeenCalledWith(true);
+
+    modelDeleteMock.mockRestore();
+    modelSearchMock.mockRestore();
+    openSearchDashboardsMock.mockRestore();
+  });
+
+  it('should show unable to delete toast and call closeModal with false after delete failed', async () => {
+    const modelDeleteMock = jest
+      .spyOn(ModelVersion.prototype, 'delete')
+      .mockRejectedValue(new Error());
+    const openSearchDashboardsMock = mockUseOpenSearchDashboards();
+    const { closeModalMock } = setup();
+
+    await userEvent.type(screen.getByLabelText('confirm text input'), 'model1 version 1');
+
+    expect(closeModalMock).not.toHaveBeenCalled();
+    await userEvent.click(screen.getByText('Delete version'));
+    expect(screen.getByTestId('euiToastHeader')).toHaveTextContent(
+      'Unable to delete model1 version 1'
+    );
+    expect(closeModalMock).toHaveBeenCalledWith(false);
+
+    modelDeleteMock.mockRestore();
+    openSearchDashboardsMock.mockRestore();
+  });
+
+  it('should call closeModal with false after delete success and still can be searched', async () => {
+    jest.useFakeTimers();
+
+    const modelDeleteMock = jest.spyOn(ModelVersion.prototype, 'delete').mockResolvedValue({});
+    const modelSearchMock = jest
+      .spyOn(ModelVersion.prototype, 'search')
+      .mockResolvedValue({ data: [], total_model_versions: 1 });
+
+    const { closeModalMock } = setup();
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    await user.type(screen.getByLabelText('confirm text input'), 'model1 version 1');
+    await user.click(screen.getByText('Delete version'));
+
+    expect(closeModalMock).not.toHaveBeenCalled();
+
+    for (let i = 0; i < 200; i++) {
```

Comment:
Seems we can't to change this. Since we are using the faker timer, the code won't be executed if we don't call `jest.advanceTimersByTime` multi times.